### PR TITLE
feat: implement CallCallablePoint to VM

### PR DESCRIPTION
**Pull request by loloicci**
# Description
This PR uses https://github.com/line/cosmwasm/pull/272 so updated after it is merged.

This PR implements CallCallablePoint to VM.
This calls a callable point (dynamic linked callee function).

This is a part of https://github.com/line/cosmwasm/issues/273.
https://github.com/line/cosmwasm/issues/273 is solved after this PR is merged and API for rust using CallCallablePoint is implemented.

## Types of changes

- [ ] Bug fix (changes which fixes an issue)
- [x] New feature (changes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] ETC (build, ci, docs, perf, refactor, style, test)

## Checklist

- [x] I followed the [contributing guidelines](https://github.com/line/wasmvm/blob/main/CONTRIBUTING.md).
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] The PR title and commits are followed [conventional commit form](https://www.conventionalcommits.org/en/v1.0.0).

### CODE REVIEW

File: libwasmvm/src/calls.rs [hunk 1]
```diff
@@ -647,3 +656,153 @@ fn do_call_3_args(
...
+        print_debug,
+    };
+    let env_u8 = env.read().ok_or_else(|| Error::unset_arg("env"))?;
+
+    // make instance
+    let mut instance = cache.get_instance(&checksum, backend, options)?;
+    instance.env.set_serialized_env(env_u8);
+    instance.set_storage_readonly(is_readonly);
+    instance.env.set_dynamic_callstack(callstack.clone())?;
+
```
Comment (loloicci) [inline libwasmvm/src/calls.rs:758]: check readonly or read-write

**Review by kokeshiM0chi — APPROVED** [review]

File: api/lib_test.go [hunk 2]
```diff
@@ -1376,5 +1376,143 @@ func TestDynamicReadWritePermission(t *testing.T) {
...
+	require.NoError(t, err)
+	argsEv := [][]byte{eventsInBin}
+	argsEvBin, err := json.Marshal(argsEv)
+	require.NoError(t, err)
+	empty := []types.HumanAddress{}
+	emptyBin, err := json.Marshal(empty)
+	require.NoError(t, err)
+
+	start = time.Now()
+	res, eventsData, attributesData, cost, err = CallCallablePoint(nameBin, cache, checksum, false, emptyBin, env, argsEvBin, &igasMeter2, store, api, &querier, TESTING_GAS_LIMIT, TESTING_PRINT_DEBUG)
```
Comment (kokeshiM0chi) [inline api/lib_test.go:1461]: Would you like to add a test case when the argument `isReadonly` = true?

File: api/lib.go [hunk 3]
```diff
@@ -606,6 +606,56 @@ func IBCPacketTimeout(
 	return copyAndDestroyUnmanagedVector(res), copyAndDestroyUnmanagedVector(events), copyAndDestroyUnmanagedVector(attributes), uint64(gasUsed), nil
 }
 
+// name: Serialized string
+// args: Serialized [][]byte
+// callstack: Serialized []string
+// returned used_gas: without instantiation cost
```
Comment (da1suk8) [inline api/lib.go:612]: Does `used_gas` refer to the name of the variable? If so, isn't it `gasUsed`?

**Review by dudong2 — APPROVED** [review]

**Review by kokeshiM0chi — CHANGES_REQUESTED** [review]
Would you like to add a test case when the argument isReadonly = true?
This has been addressed [here](https://github.com/line/wasmvm/pull/97/files#diff-41f47570e26c8c3ff485fb26d89e5fb3c5468799c6b07fe3449dd2415730fd42R1521). Thanks.

**Review by kokeshiM0chi — APPROVED** [review]

**Review by da1suk8 — APPROVED** [review]

**Comment by loloicci** [comment]
Cannot merge into dynamic_link because of https://github.com/line/wasmvm/issues/102.
All changes in this PR are included in https://github.com/line/wasmvm/pull/103.

**Pull request closed without merging**